spec/models/spree_amazon/address_spec.rb

```diff
@@ -86,42 +87,57 @@
   def stub_amazon_response(order_reference, response_data, address_consent_token: nil)
     Spree::Gateway::Amazon.create!(name: 'Amazon', preferred_test_mode: true)
     mws = instance_double(AmazonMws)
-    response = AmazonMwsOrderResponse.new(response_data)
-    allow(mws).to receive(:fetch_order_data).and_return(response)
-    allow(AmazonMws).to receive(:new).
-      with(
-        order_reference,
-        gateway: gateway,
-        address_consent_token: address_consent_token,
-      ).
-      and_return(mws)
+
+    if address_consent_token.nil?
+      body = hash_including(
+        'Action' => 'GetOrderReferenceDetails'
+      )
+    else
+      body = hash_including(
+        'Action' => 'GetOrderReferenceDetails',
+        'AddressConsentToken' => address_consent_token
+      )
+    end
```

Comment:
you can do 

```
body = if address_consent_token.nil?
  hash_including ...
else 
  hash_including..
end
```

it's also better to avoid the repetition of the `Action` as it's the same, so just create the has outside the if, and merge `ConsentToken` key if present